### PR TITLE
[WFCORE-4483] Add support to Elytron for missing MP-JWT requirements

### PR DESCRIPTION
**Pull request by nekdozjam**
https://issues.jboss.org/browse/EAP7-1259

### CODE REVIEW

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1834[ELY-1834]
+* https://issues.jboss.org/browse/WFLY-1834[WFLY-1834]
+
+=== Dev Contacts
+
+* mailto:mmazanek@redhat.com[Martin Mazanek]
+
+=== QE Contacts
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: That would be me.

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+=== Affected Projects or Components
+
+* WildFly, Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: The requirements make sense to me, but they are a bit vague. Could you be a bit more specific about the code in which the change is going to be made?

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+* change java.security.Principal returned from javax.ws.rs.core.SecurityContext.getUserPrincipal() to be instance of org.eclipse.microprofile.jwt.JsonWebToken instead of current org.wildfly.security.auth.principal.NamePrincipal
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: Is there any chance that any of the customers depend on this and it's going to break backwards compatibility?

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+=== Nice-to-Have Requirements
+
+* support injection of current token claims using org.eclipse.microprofile.jwt.Claim qualifier
+
+=== Non-Requirements
+
+== Test Plan
+
+Most of the testing will be done using existing token related tests. This issue requires only few more tests to make sure the obtained identity is of a right type and if the CDI side works.
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: Which token-related tests?

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1259[EAP7-1259]
+* https://issues.jboss.org/browse/WFCORE-4483[WFCORE-4483]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1834[ELY-1834]
+* https://issues.jboss.org/browse/WFLY-1834[WFLY-1834]
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: What's the relation here?

**Comment by sberyozkin** [comment]
Hello, as suggested by Darran, I'm copying here the list of my objections for the WildFly to pursue its own MP JWT impementation:

- Elytron based implementation of the MP JWT would defeat the purpose of creating the smallrye-jwt project in the 1st place
- It ships the code detecting the EE Security and also the JAX-RS filters
- it is the project which is moving forward - it already has much more than the MP JWT has
- it will depend in time on smallrye-jose which will let the users do the typed JWS signing calls  
- users are contributing

The proposal if for the EAP security experts to work with the smallrye-jwt team to make sure it meets the EAP quality/etc requirements

File: elytron/WFCORE-4483-microprofile-jwt.adoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+There are two main security pieces that are currently missing but will be needed for MP JWT support whenever that happens:
+
+* Support for injection of the currently authenticated caller as a JsonWebToken
+* The java.security.Principal returned from javax.ws.rs.core.SecurityContext.getUserPrincipal() must be an instance of org.eclipse.microprofile.jwt.JsonWebToken
+
+This issue focuses on adding these requirements.
+
+https://www.eclipse.org/community/eclipse_newsletter/2017/september/article2.php[Eclipse MP-JWT overview]
+
+This issue is not intended as an official support for MP-JWT, so this is not targeted towards specific MP-JWT spec version. Development will be done using version 1.1.1.
```
Comment (vobratil) [inline elytron/WFCORE-4483-microprofile-jwt.adoc]: Do you mean version 1.1.1 of MP-JWT?

**Comment by darranl** [comment]
I am going to close this proposal for now, we are presently assembling a plan that will also incorporate smallrye-jwt with a new subsystem to activate it.

**Pull request closed without merging**